modules/pool/pool.service.ts

```diff
@@ -124,6 +126,15 @@ export class PoolService {
         return featuredPoolGroups;
     }
 
+    public async getSnapshotsForAllPools(range: GqlPoolSnapshotDataRange) {
+        const allPools = await poolService.getGqlPools({});
```

Comment:
so you do this madness query with all those joins just to get the poolId? also be aware that this filters out low liquidity pools. cant you just get the snapshots directly?

Reply:
xD